pkg/virt-handler/vm.go

```diff
@@ -1362,7 +1379,7 @@ func (d *VirtualMachineController) processVmUpdate(origVMI *v1.VirtualMachineIns
 		return fmt.Errorf("failed to handle migration proxy: %v", err)
 	}
 
-	if d.isPreMigrationTarget(vmi) {
+	if d.isPreMigrationTarget(vmi) && !isMigrating(vmi.Status.MigrationState.StartTimestamp, vmi.Status.MigrationState.EndTimestamp) {
```

Comment:
As far as I remember we only set `EndTimestamp` when we detect the VM has been migrated.
Perhaps, we should check if `MigrationState.Completed` is set to indicated completion of the migration.

Reply:
I think this logic is causing failures. 

If we do get here somehow instead of calling `SyncMigrationTarget` we will always get to the `else` statement and call `SyncVirtualMachine`.

Try changing the above to: 
```suggestion
	if d.isPreMigrationTarget(vmi) && !isMigrating(vmi.Status.MigrationState.StartTimestamp, vmi.Status.MigrationState.EndTimestamp) {
```